gateway/api/middleware.go

```diff
@@ -315,10 +316,26 @@ func (a *Api) TrackRequest(eventName string) func(c *gin.Context) {
 			c.Next()
 			return
 		}
+		org, err := pgorgs.New().FetchOrgByContext(ctx)
```

Comment:
I recommend moving this to the middleware layer because we're already obtaining this information from the user (license data, org, etc). This part should only use this information contained in the `ctx` variable.